driver/src/test/unit/com/mongodb/util/JSONCallbackTest.java

```diff
@@ -139,4 +142,45 @@ public void numberDecimalParsing() {
         Decimal128 number = (Decimal128) JSON.parse(("{\"$numberDecimal\" : \"314E-2\"}"));
         assertEquals(number, Decimal128.parse("314E-2"));
     }
+
+    private Object objectDoneTestHelper(final String jsonString, final String rootName) {
```

Comment:
You can simplify the test cases by extending the existing `binaryParsing` test and add a test case similar to the existing one with a hex string eg:

```
Binary parsedBinaryWithHexType = (Binary) JSON.parse(("{ \"$binary\" : \"YWJjZA==\", \"$type\" : \"80\" }"));
assertEquals(128, parsedBinaryWithHexType.getType());
```